butane_codegen/src/lib.rs

```diff
@@ -170,8 +170,82 @@ fn migrations_dir() -> PathBuf {
 
 #[proc_macro_derive(FieldType)]
 pub fn derive_field_type(input: TokenStream) -> TokenStream {
-    let ast = syn::parse_macro_input!(input as syn::DeriveInput);
-    derive_field_type_with_json(&ast.ident)
+    let derive_input = syn::parse_macro_input!(input as syn::DeriveInput);
+    let ident = &derive_input.ident;
+    match derive_input.data {
+        syn::Data::Struct(_) => derive_field_type_with_json(ident),
+        syn::Data::Enum(data_enum) => derive_field_type_for_enum(ident, data_enum),
+        syn::Data::Union(_) => panic!("Cannot derive FieldType for a union"),
+    }
+}
+
+fn derive_field_type_for_enum(ident: &Ident, data_enum: syn::DataEnum) -> TokenStream {
+    data_enum.variants.iter().for_each(|variant| {
+        if variant.fields != syn::Fields::Unit {
+            panic!("Only simple enums are supported when deriving FieldType")
```

Comment:
can we have this failure mode instead revert back to using `derive_field_type_with_json` (if that feature is enabled) which does support complex enums?